src/renderer/components/status-bar/auto-update-component.tsx

```diff
@@ -0,0 +1,120 @@
+import type { DiscoveredUpdateVersion } from "../../../common/application-update/discovered-update-version/discovered-update-version.injectable";
+import discoveredUpdateVersionInjectable from "../../../common/application-update/discovered-update-version/discovered-update-version.injectable";
+import type { UpdateIsBeingDownloaded } from "../../../common/application-update/update-is-being-downloaded/update-is-being-downloaded.injectable";
+import updateIsBeingDownloadedInjectable from "../../../common/application-update/update-is-being-downloaded/update-is-being-downloaded.injectable";
+import type { UpdatesAreBeingDiscovered } from "../../../common/application-update/updates-are-being-discovered/updates-are-being-discovered.injectable";
+import updatesAreBeingDiscoveredInjectable from "../../../common/application-update/updates-are-being-discovered/updates-are-being-discovered.injectable";
+
+interface Dependencies {
+  progressOfUpdateDownload: ProgressOfUpdateDownload;
+  discoveredVersionState: DiscoveredUpdateVersion;
+  downloadingUpdateState: UpdateIsBeingDownloaded;
+  checkingForUpdatesState: UpdatesAreBeingDiscovered;
+}
+
+interface EndNoteProps {
+  version?: string;
+  note: (version: string) => JSX.Element;
+}
+
+const EndNote = ({ version, note }: EndNoteProps) => {
+  const [idling, setIdling] = useState(false);
+
+  useEffect(() => {
+    const timerId = setTimeout(() => setIdling(true), 5000);
+
+    return () => clearTimeout(timerId);
+  });
+
+  if (idling) {
+    return idle();
+  }
+
+  return note(version ?? "");
+};
+
+const DivWithTestId = (text: string) => <div data-testid="auto-update-component">{text}</div>;
+
+const checking = () => (
+  <>
+    <Spinner/>
+    {DivWithTestId("Checking for updates..." )}
+  </>
+);
+
+const available = (version: string) => DivWithTestId(`${version ?? "Update"} is available`);
+
+const notAvailable = () => DivWithTestId("No new updates available");
+
+const downloading = (version: string, percentDone: number) => {
+  if ( percentDone === 0 ) {
+    return (
+      <>
+        {DivWithTestId(`Download for version ${version} started...`)}
+        <Spinner/>
+      </>
+    );
+  }
+
+  return DivWithTestId(`Download for version ${version} ${percentDone}%...`);
+};
+
+const downloadFailed = (errMsg: string) => DivWithTestId(errMsg);
+
+const idle = () => DivWithTestId("");
+
+
+export const NonInjectedAutoUpdateComponent = observer(({
+  progressOfUpdateDownload,
+  discoveredVersionState,
+  downloadingUpdateState,
+  checkingForUpdatesState,
+}: Dependencies) => {
+  const discoveredVersion = discoveredVersionState.value.get();
+
+  const { percentage, failed } = progressOfUpdateDownload.value.get();
+
+  if (downloadingUpdateState.value.get()) {
+
+    assert(discoveredVersion);
+
+    const roundedPercentage = Math.round(percentage);
+
+    return downloading(discoveredVersion.version, roundedPercentage);
```

Comment:
nit: At least for me it's bit surprising that we are using functions here. Would be easier to follow if these were components. That would also give us possibility to add observers (and therefore not e.g. reload whole "auto-update-component" when only percentage changes.) But again, as long it works, I don't see any huge benefits here. I'm fine with it.